src/main/java/com/exasol/mavenprojectversiongetter/MavenProjectVersionGetter.java

```diff
@@ -24,13 +27,36 @@ private MavenProjectVersionGetter() {
 
     /**
      * Get the version of the current project.
-     * 
+     *
      * @return version string
      */
     public static String getCurrentProjectVersion() {
         return getProjectVersion(Path.of("pom.xml"));
     }
 
+    /**
+     * Get the version of the current project as specified for the project itself or for its parent.
+     *
+     * @return version string
+     */
+    public static String getVersionOfCurrentProjectOrParent() {
+        return getVersionOfProjectOrParent(Path.of("pom.xml"));
+    }
+
+    /**
+     * Get the version as specified for the project itself or for its parent
+     *
+     * @param pomFile path to {@code pom.xml} to get the version from
+     * @return version string
+     */
+    public static String getVersionOfProjectOrParent(final Path pomFile) {
+        return Stream.of(getProjectVersion(pomFile), //
+                getParentVersion(pomFile)) //
+                .filter(not(String::isEmpty)) //
+                .findFirst() //
+                .orElse("");
```

Comment:
One thing that is missing here is there is no error detection if the versions mismatch. You could do that by replacing `findFirst()` with a filter that merges duplicates and see if you still have two entries at the end. If not, everything is shiny.